### PR TITLE
Add code sniffer in travis.yml, Add gitignore, Edit readme

### PR DESCRIPTION
**Pull request by daniellevi1**
#3

### CODE REVIEW

**Comment by nirgn975** [comment]
@daniellevi1 
- Assign this PR to yourself.
- Add `WIP: ..` in the title.
- Remove the `: 3` from the title.
- Add a link to the issue (by add an hashtag and the issue number) in the PR description.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+sudo: required
+dist: trusty
+
+language: python
+python:
```
Comment (nirgn975) [inline .travis.yml]: Because we only have one version do it like this: `python: 3.5`

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  - 3.5
+
```
Comment (nirgn975) [inline .travis.yml]: There is no need for this white space, please remove one of the `enter`s.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+install:
+  # Install code sniffer.
+  - if [[ $TRAVIS_PYTHON_VERSION == 3.5 ]]; then
+    pip install pycodestyle;
+    pip install --upgrade pycodestyle;
+    source ~/.bash_profile;
+    source ~/.bashrc;
+    fi
+
```
Comment (nirgn975) [inline .travis.yml]: There is no need for this white space, please remove one of the `enter`s.

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+dist: trusty
+
+language: python
+python:
+  - 3.5
+
+
+install:
+  # Install code sniffer.
+  - if [[ $TRAVIS_PYTHON_VERSION == 3.5 ]]; then
```
Comment (nirgn975) [inline .travis.yml]: This is a script, and you don't need the `if` so you actually want something like this:

``` yml
install:
  # Install code sniffer.
  - pip install pycodestyle
  - pip install --upgrade pycodestyle
  - source ~/.bash_profile
  - source ~/.bashrc
```

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    pip install pycodestyle;
+    pip install --upgrade pycodestyle;
+    source ~/.bash_profile;
+    source ~/.bashrc;
+    fi
+
+
+script:
```
Comment (nirgn975) [inline .travis.yml:14]: Same as before, every command in a different line, like:

``` yml
script:
  # Run code sniffer on the scraper directory.
  - pycodestyle --show-source --max-line-length=119 --show-pep8 .
```

And remove `- if [[ $TRAVIS_PYTHON_VERSION == 3.5 ]]; then` and `fi`

**Review by nirgn975 — CHANGES_REQUESTED** [review]

**Comment by isranumbers** [comment]
for some inspiration:
https://www.youtube.com/watch?v=S0gMOjjH2fw

**Comment by daniellevi1** [comment]
@nirgn975 I tried to pull 2 commits to this PR and it failed because it can't build, why?
Another thing, how to delete commit in PR?
Also, 'Add a link to the issue (by add an hashtag and the issue number) in the PR description.' - it's not working to me.

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+  # Install code sniffer.
+    pip install pycodestyle;
+    pip install --upgrade pycodestyle;
+    source ~/.bash_profile;
+    source ~/.bashrc;
+
+script:
+  # Run code sniffer on the scraper directory.
+    pycodestyle --show-source --max-line-length=119 --show-pep8 .
+
```
Comment (nirgn975) [inline .travis.yml]: There is 2 `enters` at the EOF (End Of File), There is no need for such a whitespace, one is enough.

File: .travis.yml [hunk 7]
```diff
@@ -0,0 +1,17 @@
+sudo: required
+dist: trusty
+
+language: python
+python: 3.5
+
+install:
+  # Install code sniffer.
+    pip install pycodestyle;
```
Comment (nirgn975) [inline .travis.yml]: For every new line inside `install:` / `script:` / etc you will need `-` like:

``` yml
- pip install pycodestyle
- pip install --upgrade pycodestyle
- source ~/.bash_profile
- source ~/.bashrc
```

Comment (nirgn975) [reply .travis.yml]: And notice I remove the semicolons `;`.

File: .gitignore [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+
+# Windows Installer files
+*.cab
+*.msi
+*.msm
+*.msp
+
+# Windows shortcuts
+*.lnk
+
```
Comment (nirgn975) [inline .gitignore]: There is 2 enters at the EOF (End Of File), There is no need for such a whitespace, one is enough.

File: .gitignore [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
```
Comment (nirgn975) [inline .gitignore]: There is no need for such a whitespace, delete this whitespace (`enter`).

**Review by nirgn975 — CHANGES_REQUESTED** [review]

**Comment by nirgn975** [comment]
@daniellevi1 
- What do you mean by _"tried to pull 2 commits to this PR"_ ?
- There is a way to delete a commit, but it's not "best practice", so don't do this, just do your changes and do another commit. As a rule of thumb never try to get back to the past and change it (;
  
  We all made mistakes all the time, and open source means we don't have to shy about this. Our work is the way and not the end result (we don't just released the end result in the "perfect" form). Others looking at our work, help us, and correct us is a good thing, this is what great about open source.
- Revert the title to `WIP: Add code sniffer in travis.yml`. The title shouldn't contain the issue number.
- Edit your PR description by click on the pan:
  wip__3_by_daniellevi1_ _pull_request__5_ _nirgn975_open_pension_scraper
  
  And try to add a link to the issue like this:
  wip__3_by_daniellevi1_ _pull_request__5_ _nirgn975_open_pension_scraper
  
  And just press `enter`.

File: .travis.yml (hunk 7 above)
Comment (nirgn975) [reply .travis.yml]: And keep only 2 spaces and then the `-`, then another space and then the command.
Example:

``` yml
  - somethin
```

File: .travis.yml [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+    pip install --upgrade pycodestyle;
+    source ~/.bash_profile;
+    source ~/.bashrc;
+
+script:
+  # Run code sniffer on the scraper directory.
+    pycodestyle --show-source --max-line-length=119 --show-pep8 .
```
Comment (nirgn975) [inline .travis.yml]: Keep only 2 spaces, then `-`, another space and then the command, like:

``` yml
script:
  # Run code sniffer on the scraper directory.
  - pycodestyle --show-source --max-line-length=119 --show-pep8 .
```

**Review by nirgn975 — CHANGES_REQUESTED** [review]

**Comment by nirgn975** [comment]
@daniellevi1 Don't forget about edit the `README.md` file.

File: .travis.yml (hunk 10 above)
Comment (daniellevi1) [reply .travis.yml]: @nirgn975 
-  "tried to pull 2 commits to this PR" I was confiused, I ment 'tried to push 2 commits to this PR' and that failed.
-  I wanted to delete commit to make the PR more orderly

**Comment by daniellevi1** [comment]
@nirgn975
- I pushed the fix commit, but I don't see it. And all  the PR is failed. Do i need to open new PR?
- "tried to pull 2 commits to this PR" I was confiused, I ment 'tried to push 2 commits to this PR' and that failed.
- I wanted to delete commit to make the PR more orderly

**Comment by nirgn975** [comment]
@daniellevi1 The PR starting to look **great!** :+1:
Don't worry about the order of the PR itself.

Now, I have a feeling travis should work, let's check why it fail. We can go to the last `commit` and put the mouse on the red `x`:
wip__add_code_sniffer_in_travis_yml__add_gitignore__edit_readme_by_daniellevi1_ _pull_request__5_ _nirgn975_open_pension_scraper

If you click it you'll be redirect to `Travis-CI`. There you'll see the script in `.travis.yml` run. 

build__4_-_nirgn975_open_pension_scraper_-_travis_ci

First of all it look like travis managed to install everything, so that's good. And you can see the last `install` command was the command that run the `sniffer`. After it you can see that he tells us there is an error in file `./scraper/scraper/settings.py` in line 19, at the start (character 1). The error number is `E2665` and the error said `block comment should start with '# '`. Then he print the line where our code is not correct. And he try to tell us how to fix it (in this case `Separate inline comments by at least two spaces.`).

Usually you'll need to go over the output on travis, fix the errors, `commit`, and let travis run again. But in this case, those lines are options, and we don't want to make them look like regular comments, and we also want to save them there. And it's a settings file.. So let's just exclude this file, so the sniffer will not check it (`--exclude=scraper/scraper/settings.py`).

File: README.md [hunk 11]
```diff
@@ -0,0 +1,21 @@
+Open Pension Scraper
```
Comment (nirgn975) [inline README.md]: This should be in `h1`, like: `# Open Pension Scraper`

File: README.md [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+
```
Comment (nirgn975) [inline README.md]: There is no need for this whitespace.

File: README.md [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+
+The repo contains Scraper and Data from Investment firms, in use for the Open Pension project. Open Pension is a "Hasadna" project, that aimed to revealing the secrets behind the Israeli pension market.
+
+Pre Requirements
```
Comment (nirgn975) [inline README.md]: This should be an `h2`, like: `## Pre Requirements`.

File: README.md [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+
+    Make sure you have Python 3.x and pip installed.
```
Comment (nirgn975) [inline README.md]: Let's make it a bullet list, so it should be like:

``` markdown
  * Make sure you have Python 3.x and pip installed.
```

File: README.md [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+    NodeJS.
+
+Installation
```
Comment (nirgn975) [inline README.md]: This should be an `h2`, like: `## Installation`.

File: README.md [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+
+    Not yet.
```
Comment (nirgn975) [inline README.md]: There is no need for indentation here

File: README.md [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+
+Tests
```
Comment (nirgn975) [inline README.md]: This should be an `h2`, like: `## Tests`.

File: README.md [hunk 18]
```diff
@@ -0,0 +1,21 @@
...
+
+    Not yet.
+
+Contribute
```
Comment (nirgn975) [inline README.md]: This should be an `h2`, like: `## Contribute`.

File: README.md [hunk 19]
```diff
@@ -0,0 +1,21 @@
...
+    Make sure you have Python 3.x and pip installed.
+    NodeJS.
+
+Installation
+
+    Not yet.
+
+Tests
+
+    Not yet.
```
Comment (nirgn975) [inline README.md]: There is no need for indentation here

**Review by nirgn975 — CHANGES_REQUESTED** [review]

File: README.md [hunk 20]
```diff
@@ -0,0 +1,21 @@
+Open Pension Scraper
+
+
+The repo contains Scraper and Data from Investment firms, in use for the Open Pension project. Open Pension is a "Hasadna" project, that aimed to revealing the secrets behind the Israeli pension market.
+
+Pre Requirements
+
+    Make sure you have Python 3.x and pip installed.
+    NodeJS.
```
Comment (nirgn975) [inline README.md]: We don't need `NodeJS` to install this project, so remove it.

**Review by nirgn975 — CHANGES_REQUESTED** [review]

**Comment by nirgn975** [comment]
@daniellevi1 looks good! If you finish just assign the PR to me (and remove yourself), write something like "ready", and remove the `WIP:` from the title.

**Comment by nirgn975** [comment]
@daniellevi1 Strange.. the sniffer still check the `settings.py` file. lets try to put `./` at the start like:

``` yml
--exclude=./scraper/scraper/setting.py
```

Overall the PR looks good! If Travis-CI work with our little trick, that's mean you finish, just assign the PR to me (and remove yourself), write something like "ready" (as a comment), and remove the WIP: from the title.

**Comment by nirgn975** [comment]
@daniellevi1 I see it still not working. Can you find why? go over the log in Travis-CI try to figure out why this command not works. Keep in mind that in `open_pension` repo it works.
Read the path we `exclude` and the path to the file the sniffer check.

**Review by nirgn975 — APPROVED** [review]

**Comment by daniellevi1** [comment]
OMG

**Pull request merged**